ehrapy/anndata/_feature_specifications.py

```diff
@@ -27,32 +87,36 @@ def infer_feature_types(adata: AnnData, layer: str | None = None, output: Litera
         layer: The layer to use from the AnnData object. If None, the X layer is used.
         output: The output format. Choose between 'tree', 'dataframe', or None. If 'tree', the feature types will be printed to the console in a tree format.
             If 'dataframe', a pandas DataFrame with the feature types will be returned. If None, nothing will be returned. Defaults to 'tree'.
+        verbose: Whether to print warnings for uncertain feature types. Defaults to True.
+
+    Examples:
+        >>> import ehrapy as ep
+        >>> adata = ep.dt.mimic_2(encoded=False)
+        >>> ep.ad.infer_feature_types(adata)
     """
+    from ehrapy.anndata.anndata_ext import anndata_to_df
+
     feature_types = {}
 
     df = anndata_to_df(adata, layer=layer)
     for feature in adata.var_names:
-        col = df[feature].dropna()
-        majority_type = col.apply(type).value_counts().idxmax()
-        if majority_type == pd.Timestamp:
-            feature_types[feature] = DATE_TAG
-        elif majority_type not in [int, float, complex]:
-            feature_types[feature] = CATEGORICAL_TAG
-        # Guess categorical if the feature is an integer and the values are 0/1 to n-1 with no gaps
-        elif np.all(i.is_integer() for i in col) and (
-            (col.min() == 0 and np.all(np.sort(col.unique()) == np.arange(col.nunique())))
-            or (col.min() == 1 and np.all(np.sort(col.unique()) == np.arange(1, col.nunique() + 1)))
+        if (
+            FEATURE_TYPE_KEY in adata.var.keys()
+            and feature in adata.var[FEATURE_TYPE_KEY]
```

Comment:
Seem to not get it right away, what is checked here?
`feature` is in the index of `adata.var[FEATURE_TYPE_KEY]` (a pd.Series), and I think this always evaluate to `True`

Reply:
Overall, I want to check if the feature has been previously annotated and, if so, keep that annotation. But you're completely right that the specific condition always evaluates to true, so I removed it.